File: patterns-java/src/test/java/me/freedom4live/patterns/java/factory/method/NormalCatFactoryTest.java
package me.freedom4live.patterns.java.factory.method;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class NormalCatFactoryTest {

    private AbstractCatFactory catFactory;


    @BeforeEach
    void setUp() {
        catFactory = new NormalCatFactory();
    }

    @Test
    void createCat_normalCat_hasTheSameMeowForever() {
        //arrange
        Cat cat = catFactory.createCat();

        //act
        String meowResult = cat.meow();

        //assert
        for (int i = 0; i < 1000; i++) {
            assertEquals(meowResult, cat.meow());
        }
    }

}

File: patterns-kotlin/src/test/kotlin/me/freedom4live/patterns/kotlin/factory/method/NormalCatFactoryTest.kt
package me.freedom4live.patterns.kotlin.factory.method

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class NormalCatFactoryTest {

    private lateinit var catFactory: AbstractCatFactory


    @BeforeEach
    fun setUp() {
        catFactory = NormalCatFactory()
    }

    @Test
    fun createCat_createsCat_hasTheSameMeow() {
        //arrange
        val cat = catFactory.createCat()

        //act
        var previous = cat.meow()
        repeat(1000) {
            val current = cat.meow()

            //assert
            Assertions.assertEquals(previous, current)

            previous = current
        }
    }
}
